### PR TITLE
Prepare homepage loading skeleton

### PR DESCRIPTION
**Pull request by diegocasillas**
WIP. Just sharing a PoC to hear thoughts about the approach before continuing. I'm not familiar with Tailwind at all so I'm just copy-pasting.

I probably would prefer to create some components to encapsulate news, threads, user threads and favorites and define the logic, but I guess this works as well.

Just a thought: at the moment, threads height is flexible. Following this approach, there might be some _visual jumping_ around user's last posts and favorites if some threads are larger than 40px. Should we define a fixed height for threads? Or perhaps give more space between those blocks?

Open to suggestions ✌️

### CODE REVIEW

File: src/react/site/home/index.tsx [hunk 1]
```diff
@@ -56,48 +61,62 @@ function Home({ onLoad }: { onLoad: () => void }) {
...
+                          className="text-sm font-medium line-clamp-2 text-primary min-h-[40px]  hover:underline"
+                        >
+                          {thread.title}
+                        </a>
+                      </div>
+                      <div className="text-gray-400 dark:text-gray-600 text-right mt-2">
+                        {thread.createdAt}
+                      </div>
+                    </div>
+                  </div>
```
Comment (diegocasillas) [inline src/react/site/home/index.tsx]: Next step IMO would be encapsulating this in a `News` component.

File: src/react/site/home/index.tsx [hunk 2]
```diff
@@ -123,10 +142,16 @@ function Home({ onLoad }: { onLoad: () => void }) {
                   ))}
             </div>
           </div>
-          <Threads.Root className="mt-3">
-            {lastThreads?.map((thread) => {
-              return <Threads.Thread key={thread.url} {...thread} />;
-            })}
+          <Threads.Root className="mt-3 h-screen">
```
Comment (diegocasillas) [inline src/react/site/home/index.tsx:87]: A different approach would be passing the threads as prop to this component and implementing the logic for the "loading threads" there if the list is empty or undefined, but I read that you probably prefer keeping these components purely presentational and as dumb as possible.

File: src/react/site/components/threads.tsx [hunk 3]
```diff
@@ -130,9 +130,16 @@ const Thread = (props: Thread) => {
   );
 };
 
+const LoadingThread = () => {
```
Comment (jvidalv) [inline src/react/site/components/threads.tsx]: I would name it `ThreadSkeleton` to more semantically accurate.

File: src/react/site/components/threads.tsx [hunk 4]
```diff
@@ -130,9 +130,16 @@ const Thread = (props: Thread) => {
...
+  return (
+    <div className="mv-ignited--thread flex justify-between relative bg-surface first:rounded-t last:rounded-b min-h-[40px] animate-pulse" />
```
Comment (jvidalv) [inline src/react/site/components/threads.tsx]: AFAIK you can get rid of all the classNames here except `bg-surface first:rounded-t last:rounded-b min-h-[40px] animate-pulse`

File: src/react/site/home/index.tsx (hunk 1 above)
Comment (jvidalv) [reply src/react/site/home/index.tsx]: Yes, feel free to do so if you feel like it.

Said that, the news presentation is totally open to debate, I did just hack something quick (Same for threads, if there are any changes to the visuals I'm open to it).

File: src/react/site/home/index.tsx (hunk 2 above)
Comment (jvidalv) [reply src/react/site/home/index.tsx:87]: Honestly, it's a very good idea. 

The only issue, is that if we go that way, then to me feels weird that it also doesn't take care of rendering the normal threads, that's my only con.

File: src/react/site/home/index.tsx (hunk 1 above)
Comment (jvidalv) [reply src/react/site/home/index.tsx]: (And with that also the NewsSkeleton so we shave code from this page)

**Review by jvidalv — COMMENTED** [review]
> Just a thought: at the moment, threads height is flexible. Following this approach, there might be some visual jumping around user's last posts and favorites if some threads are larger than 40px. Should we define a fixed height for threads? Or perhaps give more space between those blocks?

Yes, I don't think it's a big issue for the main block of threads, but could be weird/ugly for favorites/last posts as they ar tighter.

I also think that threads going over two lines creates a weird effect. But I wouldn't fix the height, instead I would clamp the text to never go over two lines (line-clamp in tailwind class), and then to not cut the threads in the right column short, we could configure the `Thread` component to remove some information, for example for last posts we could remove total responses and timestamp, and made them look like favorites.

With these changes, now the skeletons and the threads should match 1:1, and then the problem left is the number in last posts and favorites not matching the skeletons, but we can tackle this after, wdyt?

(Also an empty state at least for favourites is required at some point)

**Comment by diegocasillas** [comment]
Ready for review! I like to make PRs as atomic as possible, that's why the minimal changes. I might prepare a following PR with a suggestion that I think might help with the initial blinking, but I need to try it out.

Let me know what you think of those "list" components. This pattern is a bit unfamiliar to me as I'm more used to work with styled components and usually I have "manager components" that encapsulate the logic and fetching.

**Review by jvidalv — APPROVED** [review]

**Comment by jvidalv** [comment]
> Ready for review! I like to make PRs as atomic as possible, that's why the minimal changes. I might prepare a following PR with a suggestion that I think might help with the initial blinking, but I need to try it out.
> 
> Let me know what you think of those "list" components. This pattern is a bit unfamiliar to me as I'm more used to work with styled components and usually I have "manager components" that encapsulate the logic and fetching.

Looks good! thanks!

Also for the initial blinking, there are a few things, first the extension injecting the JS supposedly (at least on chrome) even before the page is fully served, but not working. Check the background.ts for that.

You could play with all these variables to see how the blinking/injecting/loading could be improved. Feel free!

**Pull request merged**